layouts/shortcodes/dropdown.html

```diff
@@ -31,23 +31,152 @@
+{{< /dropdown >}}
+
+{{< dropdown >}}
+title = 'Custom chevron with animation'
+icon = 'fa-solid fa-star'
+chevron = 'down-up'
+animate = 'fade-in'
+color = 'warning'
+body = 'Combining different features!'
+{{< /dropdown >}}
+
+{{< dropdown >}}
+title = 'Multi-line content'
+icon = 'fa-solid fa-align-left'
+body = '''
+This is multi-line content.
+
+It can include **markdown** and multiple paragraphs.
+And even lists:
+
+- Item 1
+- Item 2
+- Item 3
+'''
+{{< /dropdown >}}
+
+{{< dropdown >}}
+title = 'Dropdown with code'
+icon = 'fa-solid fa-code'
+color = 'success'
+body = '''
+Here's some code inside a dropdown:
+
+```python
+def hello_world():
+    print("Hello from a dropdown!")
+    return 42
+```
+
+And more content after the code.
+'''
+{{< /dropdown >}}
+
+{{< dropdown >}}
+title = 'Nested dropdowns'
+icon = 'fa-solid fa-folder'
+body = '''
+This is a dropdown with a dropdown inside it.
+
+{{< dropdown >}}
+title = 'Inner dropdown'
+icon = 'fa-solid fa-code'
+body = '''
+I'm inside a dropdown in a dropdown!
+'''
+{{< /dropdown >}}
+
+And here is some more content after the inner dropdown.
+'''
+{{< /dropdown >}}
+
 */}}
 
 {{- $data := .Inner | transform.Unmarshal -}}
-<details class="sd-card sd-dropdown sd-mb-3">
-  {{- with $data.color }}
-  <summary class="sd-summary-title sd-card-header sd-bg-{{ . }} sd-bg-text-{{ . }}">
-  {{- else }}
-    <summary class="sd-summary-title sd-card-header">
-  {{- end }}
-  {{- with $data.icon }}
-  <span class="sd-summary-icon"><i class="fa {{ . }}"></i></span>
-  {{- end }}
-  {{- with $data.title }}
-  {{- . }}
-  {{- else }}
-  <i class="fas fa-ellipsis-h"></i>&nbsp;
-  {{- end }}
+
+{{- $chevronClass := "sd-summary-chevron-right" -}}
+{{- $chevronIcon := "fa-chevron-right" -}}
+{{- if eq $data.chevron "down-up" -}}
+  {{- $chevronClass = "sd-summary-chevron-down" -}}
+  {{- $chevronIcon = "fa-chevron-down" -}}
+{{- end -}}
+
+{{- $detailsClasses := slice "sd-sphinx-override" "sd-dropdown" "sd-card" "sd-mb-3" -}}
+{{- with $data.animate -}}
+  {{- $detailsClasses = $detailsClasses | append (printf "sd-%s" .) -}}
+{{- end -}}
+
+<details class="{{ delimit $detailsClasses " " }}"{{ if $data.open }} open="open"{{ end }}>
+  {{- $summaryClasses := slice "sd-summary-title" "sd-card-header" -}}
+  {{- with $data.color -}}
+    {{- $summaryClasses = $summaryClasses | append (printf "sd-bg-%s" .) -}}
+    {{- $summaryClasses = $summaryClasses | append (printf "sd-bg-text-%s" .) -}}
+  {{- end -}}
+
+  <summary class="{{ delimit $summaryClasses " " }}">
+    {{- with $data.icon }}
+    <span class="sd-summary-icon"><i class="fa {{ . }}"></i></span>
+    {{- end }}
+    <span class="sd-summary-text">
+    {{- with $data.title }}
+      {{- . }}
+    {{- else }}
+      <i class="fas fa-ellipsis-h no-title sd-octicon"></i>
+    {{- end }}
+    </span>
+    <span class="sd-summary-state-marker {{ $chevronClass }}">
+      <i class="fas {{ $chevronIcon }}"></i>
+    </span>
```

Comment:
`.sd-summary-state-marker` is a separate element that makes it easier for us to style where the chevrons are placed (at the end of the dropdown bars).

Reply:
`sd-` belongs to Sphinx Design, so is this prefix intentional?